### ISSUE TITLE
Providers Broken via MIssing Type Name

### ISSUE THREAD

**Issue by RobbieMcKinstry** [pulumi/pulumi#10515]
### What happened?

**TL;DR:** [This commit](https://github.com/pulumi/pulumi/pull/10435/files) has introduced a change which has broken provider codegen for at least `@pulumi/command`.

I recently checked out `@pulumi/command` to see work on codegen changes. I noticed that codegen was not working, and that the error seemed legitimate. 

I checked out Pulumi v3.38.0, and I confirmed that codegen works fine. I then used `git bisect` and determined this above commit was the offending comment, which jives with the error message from `@pulumi/command`:

```
make build
(cd provider && go build -o /Users/robbiemckinstry/workspace/pulumi/pulumi-command/bin/pulumi-gen-command -ldflags "-X github.com/pulumi/pulumi-command/provider/pkg/version.Version=0.4.2-alpha.1661516193+2d1657e1.dirty" github.com/pulumi/pulumi-command/provider/cmd/pulumi-gen-command)
# github.com/pulumi/pulumi/pkg/v3/resource/deploy/deploytest
../../pulumi/pkg/resource/deploy/deploytest/pluginhost.go:423:57: undefined: workspace.ProjectPlugin
# github.com/pulumi/pulumi/pkg/v3/engine
../../pulumi/pkg/engine/plugins.go:180:99: undefined: workspace.ProjectPlugin
make: *** [codegen] Error 2
```

It's not clear to me _why_ `deploytest` can't resolve `workspace.ProjectPlugin` in `pulumi/command`, since `workspace` is already imported and  [ProjectPlugin](https://github.com/pulumi/pulumi/pull/10435/files#diff-196f5b7811e033d2523373f8138cca8c54eba850b419591283cd88a35828ac12R510-R515) seems to be defined.

### Steps to reproduce

1. Check out the offending commit, or any later commit.
2. Check out `@pulumi/command`.
3. Update `provider/go.mod` in pulumi/command to point to your local pulumi/pulumi repo.
4. Run `make ensure && make build` in `@pulumi/command`
5. Observe the error message.

### Expected Behavior

I expected `pulumi/command` to successfully codegen.

### Actual Behavior

See above error message.

### Output of `pulumi about`

```
CLI          
Version      3.38.0
Go Version   go1.19
Go Compiler  gc

Host     
OS       darwin
Version  12.3
Arch     arm64

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/thesnowmancometh
User           thesnowmancometh
Organizations  thesnowmancometh, whiterabbit, pulumi

Pulumi locates its logs in /var/folders/8n/y4s73s2d4rnbn7d85clz9pz40000gn/T/ by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Frassle** [comment]
Did you replace pkg and sdk? Because trying to build HEAD of pkg against 3.38.0 sdk is expected to error (because of this api change).

**Comment by RobbieMcKinstry** [comment]
I replaced just `pkg`. Let me try to also replace sdk.

**Comment by RobbieMcKinstry** [comment]
You should hear me cursing. This fixed it for me.

I'm making a note to update our documentation with these instructions.

**Comment by stack72** [comment]
> Did you replace pkg and sdk? Because trying to build HEAD of pkg against 3.38.0 sdk is expected to error (because of this api change).

What new API change? That feels like we should talk about that type of change before shipping as it’ll heavily affect us

**Comment by Frassle** [comment]
> What new API change? 

The one linked at the top of the ticket https://github.com/pulumi/pulumi/pull/10435/files. We split some workspace plugin code up.

> That feels like we should talk about that type of change before shipping as it’ll heavily affect us

It shouldn't, I had a look around and the only affected appeared to be tfbridge, and it was a very small fix https://github.com/pulumi/pulumi-terraform-bridge/pull/595. you could even prepare for the fix without updating the SDK dependency by just removing the downloadurl line shown in that pr.
